tubemq-server/src/main/java/org/apache/tubemq/server/broker/msgstore/disk/MsgFileStore.java

```diff
@@ -169,8 +169,10 @@ public void batchAppendMsg(final StringBuilder sb, final int msgCnt,
             }
             // check whether need to flush to disk.
             long currTime = System.currentTimeMillis();
+            int unflushDataHold = this.messageStore.getTopicMetadata().getUnflushDataHold();
```

Comment:
The topicMetadata initialized may not be the latest metadata object. It should be safe to construct the unflushdatahold according to the unflushinterval